internal/inspect/utils.go

```diff
@@ -0,0 +1,68 @@
+package inspect
+
+import (
+	"embed"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/go-errors/errors"
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/supabase/cli/internal/db/reset"
+	"github.com/supabase/cli/internal/utils"
+)
+
+const ROOT_DIR = "queries/"
+const CSV_QUERY = `COPY (%s) TO STDOUT WITH CSV HEADER`
+
+//go:embed queries/*.sql
+var queries embed.FS
+
+func ReadQuery(query string) string {
+	path := fmt.Sprintf("%s%s.sql", ROOT_DIR, query)
+	queryString, err := queries.ReadFile(path)
+	if err != nil {
+		println(err.Error())
+		return ""
+	}
+	return string(queryString)
+}
+
+func Report(cmd *cobra.Command, config pgconn.Config, fsys afero.Fs, options ...func(*pgx.ConnConfig)) error {
+	queries, err := queries.ReadDir("queries")
+	if err != nil {
+		return err
+	}
+	date := time.Now().Format("2006-01-02")
+	out, err := cmd.Flags().GetString("out-path")
+	if err != nil {
+		return err
+	}
+	if out != "" {
+		err := os.MkdirAll(out, os.ModePerm)
+		if err != nil {
+			return errors.Errorf("failed to create output directory: %w", err)
+		}
+	} else {
+		out = "./"
+	}
+	fmt.Println("Running queries...")
+	for _, v := range queries {
+		name := strings.Split(v.Name(), ".")[0]
+		query := ReadQuery(name)
+		fq := strings.Replace(query, "$1", "'{"+strings.Join(reset.LikeEscapeSchema(utils.InternalSchemas), ",")+"}'::text[]", -1)
+		copyCmd := fmt.Sprintf(`COPY (%s) TO STDOUT WITH CSV HEADER`, fq)
+		// gosec lint error is unavoidable here because of formatted params
+		cmd := exec.CommandContext(cmd.Context(), "psql", utils.ToPostgresURL(config), "-At", "-F\",\"", "-c", copyCmd, "-o", fmt.Sprintf("%s/%s_%s.csv", out, name, date)) //nolint:gosec
```

Comment:
Let's not introduce a binary dependency on psql. Ideally we want to use pgx library for consistency with all other commands.

Reply:
That was my first efforts as well. Unfortunately, it is not possible for `pgx` to do this. `COPY` is only supported TO the database and not from it.
Potentially the `database/sql` module supports it but I did not see it.

The other option, which I not keen on, is to convert the output from pgx to CSV and then direct to a file

Reply:
> The other option, which I not keen on, is to convert the output from pgx to CSV and then direct to a file

What's your concern with this approach?

IMO, this is the nicer option as reports would run queries exactly the same way as other inspect commands. It's more testable and we can avoid doing our own string substitution for $1 template variable.

Reply:
Pros:
- type safety
- more control over output and errors
Cons:
- cost to implement is prohibitive (I am unlikely to do it 😅 )
- change to queries impacts output
- not the "postgres" way as we are wrapping query output that is already formatted

Reply:
I can look into implementing it using pgx since I'm more familiar with the codebase. Probably after reliability improvements.

Reply:
welp, I was wrong. I saw v4 dropped `CopyToWriter` so I assumed it was no longer supported. Pushed an update to drop `psql` requirement, sorry about that!